packages/signature-v4/src/SignatureV4.ts

```diff
@@ -291,19 +297,42 @@ ${toHex(hashedRequest)}`;
     return toHex(await hash.digest());
   }
 
-  private getSigningKey(credentials: Credentials, region: string, shortDate: string): Promise<Uint8Array> {
-    return getSigningKey(this.sha256, credentials, shortDate, region, this.service);
+  private getSigningKey(
+    credentials: Credentials,
+    region: string,
+    shortDate: string,
+    service?: string
+  ): Promise<Uint8Array> {
+    return getSigningKey(this.sha256, credentials, shortDate, region, service || this.service);
   }
 }
 
-function formatDate(now: DateInput): { longDate: string; shortDate: string } {
+const formatDate = (now: DateInput): { longDate: string; shortDate: string } => {
   const longDate = iso8601(now).replace(/[\-:]/g, "");
   return {
     longDate,
     shortDate: longDate.substr(0, 8),
   };
-}
+};
 
-function getCanonicalHeaderList(headers: object): string {
-  return Object.keys(headers).sort().join(";");
-}
+const getCanonicalHeaderList = (headers: object): string => Object.keys(headers).sort().join(";");
+
+const normalizeRegionProvider = function (region: string | Provider<string>): Provider<string> {
```

Comment:
nit: arrow function
```suggestion
const normalizeRegionProvider = (region: string | Provider<string>): Provider<string> => {
```

Ditto on line 329

Reply:
I used function here because I thought the return value `() => promisified` will 'memoized'. But it's not the case. Will update it.